src/exporters/s5.py

```diff
@@ -0,0 +1,492 @@
+        min_month: int = 1,
+        max_month: int = 12,
+        max_leadtime: Optional[int] = None,
+        pressure_levels: Optional[List[int]] = None,
+        selection_request: Optional[Dict] = None,
+        N_parallel_requests: int = 3,
+        show_api_request: bool = True,
+    ):
+        """
+        Arguments
+        --------
+        min_year: Optional[int] default = 2017
+            the minimum year of your request
+
+        max_year: Optional[int] default = 2018
+            the maximum year of your request
+
+        min_month: Optional[int] default = 1
+            the minimum month of your request
+
+        max_month: Optional[int] default = 12
+            the maximum month of your request
+
+        max_leadtime: Optional[int]
+            the maximum leadtime of your request
+                (if granularity is `hourly` then provide in days)
+                (elif granularity is `monthly` then provide in months)
+            defaults to ~3 months (90 days)
+
+        pressure_levels: Optional[int]
+            Pressure levels to download data at
+
+        Note:
+        ----
+        - All parameters that are assigned to class attributes are fixed for one download
+        - these are required to initialise the object [granularity, pressure_level]
+        - Only time will be chunked (by months) to send separate calls to the cdsapi
+        """
+        # N_parallel_requests can only be a MINIMUM of 1
+        if N_parallel_requests < 1:
+            N_parallel_requests = 1
+
+        # max_leadtime defaults
+        if max_leadtime is None:
+            # set the max_leadtime to 3 months as default
+            max_leadtime = 90 if (self.granularity == "hourly") else 3
+
+        if pressure_levels is None:
+            # set the pressure_levels to ['200', '500', '925'] as default
+            pressure_levels = [200, 500, 925] if (self.pressure_level) else None
+
+        processed_selection_request = self.create_selection_request(
+            variable=variable,
+            max_leadtime=max_leadtime,
+            min_year=min_year,
+            max_year=max_year,
+            min_month=min_month,
+            max_month=max_month,
+            selection_request=selection_request,
+        )
+
+        if self.pressure_level:  # if we are using the pressure_level dataset
+            pressure_levels_dict = self.get_pressure_levels(pressure_levels)
+
+            if pressure_levels_dict is not None:
+                processed_selection_request.update(cast(Dict, pressure_levels_dict))
+
+        if N_parallel_requests > 1:  # Run in parallel
+            # p = multiprocessing.Pool(int(N_parallel_requests))
+            p = pool(int(N_parallel_requests))  # pathos seems to pickle classes
+
+        # SPLIT THE API CALLS INTO MONTHS (speed up downloads)
+        output_paths = []
+        for year, month in itertools.product(
+            processed_selection_request["year"], processed_selection_request["month"]
```

Comment:
Nice this is clever. Should it be optional though, using a `break_up` input argument?

Reply:
haha this is your code 👍 (i agree its v.clever)

Do we definitely want this to be an option? This is one of teh things i think we probably want to assume for the user?

Although saying that you might be right because i don't think its the size of the data download limiting speedds. i think i'm somehow at the bottom of the food chain

Reply:
okay added it in as an option

Reply:
might be interesting to test running this in parallel outside of the class for example